### DISCUSSION TITLE
Absolute Encoder Bank

### DISCUSSION THREAD

**Discussion by Zlraul** [tttapa/Control-Surface#532] [Q&A] [answered]
hey, I need to use an encoder for my midi stuff and I need to bank it, but it keeps giving me error message "invalid use of template-name 'CS::Bankable::CCAbsoluteEncoder' without an argument list".
 I already try to change some parts but nothing works, the only thing that works is changing the "CCAbsoluteEncoder" to "CCRotaryEncoder" but that gave me just a value from 0 to 127 with nothing in between.

can anyone help me with a bank using CCAbsoluteEncoder?

Thanks.

this is the code:

```
#include <Control_Surface.h>

USBMIDI_Interface midi;
USBDebugMIDI_Interface midiInterface(115200);

Bank<2> Nbank(3);
Bank<2> Pbank(1);
Bank<2> Rbank(1);

Bankable::CCPotentiometer knobsSide[] {
  {{Pbank, BankType::CHANGE_ADDRESS}, A0, {MIDI_CC::Effect_Control_1, CHANNEL_3}},
  //{{bank1, BankType::CHANGE_ADDRESS}, A1, {MIDI_CC::General_Purpose_Controller_2_LSB, CHANNEL_1}},
  };

Bankable::NoteButton Buttons[] {
  {{Nbank, BankType::CHANGE_ADDRESS}, 3, {MIDI_Notes::C(3), CHANNEL_3}},
  {{Nbank, BankType::CHANGE_ADDRESS}, 4, {MIDI_Notes::Db(3), CHANNEL_3}},
  {{Nbank, BankType::CHANGE_ADDRESS}, 5, {MIDI_Notes::D(3), CHANNEL_3}},
};

Bankable::CCAbsoluteEncoder enc[]{
  {{Rbank, BankType::CHANGE_ADDRESS}, {6, 7}, MIDI_CC::General_Purpose_Controller_7, 1 },
};

SwitchSelector Slc {Pbank, 2};
SwitchSelector Slc1 {Nbank, 2};
SwitchSelector Slc2 {Rbank, 2};

void setup(){
  Control_Surface.begin();
}

void loop(){
  Control_Surface.loop();
}

```

**Comment by tttapa** [comment, marked as answer]
[`Bankable::CCAbsoluteEncoder`](https://tttapa.github.io/Control-Surface-doc/Doxygen/d7/d2c/classBankable_1_1CCAbsoluteEncoder.html) is a class template. You have to provide the number of banks as an argument, which is why the compiler complains.

```cpp
Bankable::CCAbsoluteEncoder<2> enc[] { ... };
```

**Reply by Zlraul** [reply]
oh it works, thank you so so much.
it such a small detail and hard to figure out (at least for a beginner like me).

once again thank you so much.
